src/lib/sql/sql_translator.cpp

```diff
@@ -128,1182 +174,1113 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::translate_statement(const hsql::
+          }
+          sql_identifier_context->set_table_name(column_expression, hsql_table_ref.name);
+        }
+
+        Assert(_use_mvcc == UseMvcc::No || lqp_is_validated(view->lqp),
+               "Can't use unvalidated View in validated Query");
+      } else {
+        FailInput(std::string("Did not find a table or view with name ") + hsql_table_ref.name);
       }
-    }
-  }
+      table_name = hsql_table_ref.alias ? hsql_table_ref.alias->name : hsql_table_ref.name;
+    } break;
 
-  if (is_aggregate) {
-    current_result_node = _translate_aggregate(select, current_result_node);
-  } else {
-    current_result_node = _translate_projection(*select.selectList, current_result_node);
+    case hsql::kTableSelect: {
+      AssertInput(hsql_table_ref.alias && hsql_table_ref.alias->name, "Every SubSelect must have its own alias");
+      table_name = hsql_table_ref.alias->name;
+
+      SQLTranslator sub_select_translator{_use_mvcc};
+      lqp = sub_select_translator.translate_select_statement(*hsql_table_ref.select);
+
+      for (const auto& sub_select_expression : lqp->column_expressions()) {
+        const auto identifier =
+            sub_select_translator.sql_identifier_context()->get_expression_identifier(sub_select_expression);
+
+        // Make sure each column from the SubSelect has a name
+        if (identifier) {
+          sql_identifier_context->set_column_name(sub_select_expression, identifier->column_name);
+        } else {
+          sql_identifier_context->set_column_name(sub_select_expression, sub_select_expression->as_column_name());
+        }
+      }
+
+      table_name = hsql_table_ref.alias->name;
+    } break;
+
+    default:
+      Fail("_translate_table_origin() is only for Tables, Views and Sub Selects.");
   }
 
-  Assert(select.unionSelect == nullptr, "Set operations (UNION/INTERSECT/...) are not supported yet");
+  // Rename columns as in "SELECT * FROM t AS x (y,z)"
+  if (hsql_table_ref.alias && hsql_table_ref.alias->columns) {
+    const auto& column_expressions = lqp->column_expressions();
+
+    AssertInput(hsql_table_ref.alias->columns->size() == column_expressions.size(),
+           "Must specify a name for exactly each column");
 
-  if (select.order != nullptr) {
-    current_result_node = _translate_order_by(*select.order, current_result_node);
+    for (auto column_id = ColumnID{0}; column_id < hsql_table_ref.alias->columns->size(); ++column_id) {
+      sql_identifier_context->set_column_name(column_expressions[column_id],
+                                              (*hsql_table_ref.alias->columns)[column_id]);
+    }
   }
 
-  // TODO(anybody): Translate TOP.
-  if (select.limit != nullptr) {
-    current_result_node = _translate_limit(*select.limit, current_result_node);
+  for (const auto& expression : lqp->column_expressions()) {
+    sql_identifier_context->set_table_name(expression, table_name);
   }
 
-  return current_result_node;
+  return {lqp,
+          {{
```

Comment:
This looks like some constructor interface can be improved

Reply:
will look into it